javascript/code-challenges/challenges-10.test.js

```diff
@@ -0,0 +1,336 @@
+'use strict';
+
+const { val } = require("cheerio/lib/api/attributes");
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named returnTen, takes in a string and uses split and splice to return the last 10 characters from that string as elements of an array.
+
+------------------------------------------------------------------------------------------------ */
+
+function returnTen(str){
+  return str.split('').splice(-10);
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named findMax that takes in a matrix of positive numbers and returns the number with the highest value.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 6],
+  [23, 5, 5]
+]
+
+return: 23
+------------------------------------------------------------------------------------------------ */
+const findMax = (matrix) => {
+ let value = 0;
+
+ for(let i=0; i<matrix.length; i++){
+   for(let k=0; k<matrix[i].length; k++){
+     if(value < matrix[i][k]){
+       value = matrix[i][k];
+     }
+   }
+ }
```

Comment:
So this doesn't matter currently, because you don't have a huge data set, but you're better off avoiding loops within loops whenever possible, they're pretty inefficient. What you can do is combine all your arrays and then use reduce, which would give you a more efficient solution.

edit: Later, when you get into 401, you'll start worrying more about the efficiency of your algorithms. This method, a loop within a loop, is often referred to as brute force and it's usually the least desirable solution to an algorithm problem.